From e3db6b8993682f2ae21fabc3c6683d48203ad253 Mon Sep 17 00:00:00 2001
From: Peter Tworek <tworaz666@gmail.com>
Date: Sun, 22 Mar 2015 18:10:50 +0100
Subject: [PATCH 5/9] Add Qt global shared context to GPU thread share group.

This allows Qt to access GL resources produced by chromium GPU thread.

Signed-off-by: Peter Tworek <tworaz666@gmail.com>
---
 content/common/gpu/gpu_channel_manager.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/content/common/gpu/gpu_channel_manager.cc b/content/common/gpu/gpu_channel_manager.cc
index 9fa18c1..a84d7e9 100644
--- a/content/common/gpu/gpu_channel_manager.cc
+++ b/content/common/gpu/gpu_channel_manager.cc
@@ -26,6 +26,10 @@
 #include "ui/ozone/public/ozone_platform.h"
 #endif
 
+#if defined(USE_QT)
+#include "ui/gl/gl_share_context_qt.h"
+#endif
+
 namespace content {
 
 namespace {
@@ -193,6 +197,9 @@ void GpuChannelManager::OnEstablishChannel(int client_id,
   if (share_context) {
     if (!share_group_.get()) {
       share_group_ = new gfx::GLShareGroup;
+#if defined(USE_QT)
+      new gfx::GLShareContextQt(share_group_.get());
+#endif
       DCHECK(!mailbox_manager_.get());
       mailbox_manager_ = new gpu::gles2::MailboxManagerImpl;
     }
-- 
2.0.5

